### PR TITLE
 No badges will show to be assigned

### PR DESCRIPTION
**Pull request by johny4437**
# Description

Fixes 11 ( BADGES BUGS - PRIORITY - MEDIUM )
 ‘Welcome, Username’ > View Profile > Featured Badges > Assign Badges
If a user hasn’t visited: Other Links > Badge Management yet, no badges will show to be assigned.

## Mainly changes explained:
1 - Added a function called `allBadges( )` that load all badges to be assigned in the component AssignBadgePopup.jsx
2 - Called this function inside `useEffect( )` to load all badges when the modal opens.
…

## How to test:
check into current branch 
do `npm install` and ` npm run start:local` to run this PR locally
log as admin user then:
‘Welcome, Username’ → View Profile →Featured Badges → Assign Badges
See the bagdes will load.
Captura de Tela (123)

https://user-images.githubusercontent.com/39320057/211075831-665f3e5b-2c43-46a8-94fc-970317d16819.mp4

### CODE REVIEW

**Review by Aishwaryak01 — COMMENTED** [review]
The code looks good to me and is able to assign a badge, however after clicking on 'confirm' to assign a badge, an error occurs.

File: src/components/UserProfile/AssignBadgePopup.jsx [hunk 1]
```diff
@@ -31,25 +32,35 @@ const AssignBadgePopup = (props) => {
...
     props.close();
   };
+  useEffect(() => {
+    loadAllBadges();
+  }, []);
 
-  const filterBadges = (allBadges) => {
-    let filteredList = allBadges.filter((badge) => {
+  const loadAllBadges = async () => {
+    try {
```
Comment (LeeSingChanYau) [inline src/components/UserProfile/AssignBadgePopup.jsx:40]: code looks clean.

**Review by LeeSingChanYau — COMMENTED** [review]
The code looks clean. But I see an error as well when pressing the confirm button. I was able to assign the badge, but let me know if you see an error as well when saving.

File: src/components/UserProfile/AssignBadgePopup.jsx (hunk 1 above)
Comment (LeeSingChanYau) [reply src/components/UserProfile/AssignBadgePopup.jsx:40]: Screenshot 2023-01-06 at 11 46 29 PM

Comment (LeeSingChanYau) [reply src/components/UserProfile/AssignBadgePopup.jsx:40]: This is the error I see after assigning the badges.

Comment (johny4437) [reply src/components/UserProfile/AssignBadgePopup.jsx:40]: Hi guys,  I tested my code here again, and I am not getting this error, take a look at the video that I did

https://user-images.githubusercontent.com/39320057/211143301-75e5d06e-e85b-43b9-ba0e-46dc46e38a8b.mp4

**Review by Aishwaryak01 — APPROVED** [review]
Tested and working as expected

**Comment by one-community** [comment]
Branch has conflicts and can’t be merged.

**Review by rauleffting — APPROVED** [review]
Were made small and clean changes in the code. Everything looks great. I tested it here and worked fine!

https://user-images.githubusercontent.com/29555732/212354439-f661699e-d36e-4aec-83c7-a458a4a6f837.mov

**Pull request merged**